### DISCUSSION TITLE
Beta Crypto API

### DISCUSSION THREAD

**Discussion by sanko** [sanko/Robinhood#95] [General]
AFAIK, they don't have a place to report issues with their API docs so I'm putting problems I find here for now.

**Comment by sanko** [comment]
You can trigger malformed responses from their backend by sending something mildly unexpected. For example, [request the estimated price of ETH with a quantity below their minimum order size](https://trading.robinhood.com/api/v1/crypto/marketdata/estimated_price/?symbol=ETH-USD&side=both&quantity=0.00001431). You'll need access to generate your own app key, etc.

You'll receive an `HTTP 400` with the following json:

    {"type":"validation_error","errors":[{"detail":"Quantity requested is too small. Min quantity for ETH-USD: 0.0001","attr":"quantity"}]}{"results":[{"symbol":"ETH-USD","price":"2999.5","quantity":"0.00001431","side":"bid","updated_at":"2024-05-08T12:20:50.337250082-04:00"},{"symbol":"ETH-USD","price":"3020.46","quantity":"0.00001431","side":"ask","updated_at":"2024-05-08T12:20:50.337250082-04:00"}]}

The actual results (even with my bad quantity value) are tacked onto the error (which should end the API's response) which makes the entire JSON response malformed.

**Comment by sanko** [comment]
Endpoint for gathering orders doesn't include `time_in_force` for limit orders even if it's set when the order is placed. Docs say it should be there inside the `limit_order_config`.

**Comment by sanko** [comment]
I feel like I'm missing something but the backend is way too sensitive to timestamp values. My clock is NTP synced but Robinhood's CloudFront instance is 5 seconds behind so it rejects requests occasionally even though the docs say [they'll accept timestamps up to 30 seconds behind their system clock](https://docs.robinhood.com/crypto/trading/#section/Authentication/Timestamp).

I'd appreciate an endpoint to gather their system time but I'm using HTTP response headers to try and find it for now.

**Reply by leep2187** [reply]
I found that my day lights savings time settings was messing with this.

**Comment by sanko** [comment]
`nummus` allowed you to place market orders based on a dollar amount (to trade $100 of BTC regardless of how much crypto that is) but the new API has no support for the equivalent (would be `market_order_config.quote_amount`). System rejects such orders.
